tests/foreman/api/test_activationkey.py

```diff
@@ -50,470 +50,462 @@ def _bad_max_hosts():
     return [gen_integer(-100, -1), 0, gen_string('alpha')]
 
 
-class ActivationKeyTestCase(APITestCase):
-    """Tests for the ``activation_keys`` path."""
+@pytest.mark.tier1
+def test_positive_create_unlimited_hosts():
+    """Create a plain vanilla activation key.
 
-    @pytest.mark.tier1
-    def test_positive_create_unlimited_hosts(self):
-        """Create a plain vanilla activation key.
+    :id: 1d73b8cc-a754-4637-8bae-d9d2aaf89003
 
-        :id: 1d73b8cc-a754-4637-8bae-d9d2aaf89003
+    :expectedresults: Check that activation key is created and its
+        "unlimited_hosts" attribute defaults to true.
 
-        :expectedresults: Check that activation key is created and its
-            "unlimited_hosts" attribute defaults to true.
+    :CaseImportance: Critical
+    """
+    assert entities.ActivationKey().create().unlimited_hosts is True
 
-        :CaseImportance: Critical
-        """
-        self.assertTrue(entities.ActivationKey().create().unlimited_hosts)
 
-    @pytest.mark.tier1
-    def test_positive_create_limited_hosts(self):
-        """Create an activation key with limited hosts.
+@pytest.mark.tier1
+@pytest.mark.parametrize('max_host', **parametrized(_good_max_hosts()))
+def test_positive_create_limited_hosts(max_host):
+    """Create an activation key with limited hosts.
 
-        :id: 9bbba620-fd98-4139-a44b-af8ce330c7a4
+    :id: 9bbba620-fd98-4139-a44b-af8ce330c7a4
 
-        :expectedresults: Check that activation key is created and that hosts
-            number is limited
+    :expectedresults: Check that activation key is created and that hosts
+        number is limited
 
-        :CaseImportance: Critical
-        """
-        for max_host in _good_max_hosts():
-            with self.subTest(max_host):
-                act_key = entities.ActivationKey(
-                    max_hosts=max_host, unlimited_hosts=False
-                ).create()
-                self.assertEqual(act_key.max_hosts, max_host)
-                self.assertFalse(act_key.unlimited_hosts)
+    :CaseImportance: Critical
+    """
+    act_key = entities.ActivationKey(max_hosts=max_host, unlimited_hosts=False).create()
+    assert act_key.max_hosts == max_host
+    assert act_key.unlimited_hosts is False
 
-    @pytest.mark.tier1
-    def test_positive_create_with_name(self):
-        """Create an activation key providing the initial name.
 
-        :id: 749e0d28-640e-41e5-89d6-b92411ce73a3
+@pytest.mark.tier1
+@pytest.mark.parametrize('key_name', **parametrized(valid_data_list()))
```

Comment:
Missing testimony token for parametrized, @latran made a general comment about including these.

Reply:
added